core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -407,8 +416,11 @@ private void updateFlags(int flags) {
 
             if (flagsUpdater.compareAndSet(this, oldFlags, newFlags)) {
                 final RequestLogFuture[] satisfiedFutures;
-                synchronized (pendingFutures) {
-                    satisfiedFutures = removeSatisfiedFutures();
+                lock.lock();
+                try {
+                    satisfiedFutures = removeSatisfiedFutures(pendingFutures, flags);
```

Comment:
```suggestion
                    satisfiedFutures = removeSatisfiedFutures(pendingFutures, newFlags);
```
I believe that this is the reason of test failures.